packages/disputer/index.js

```diff
@@ -46,26 +54,62 @@ async function run({
 }) {
   try {
     const { toBN } = web3.utils;
-
     const getTime = () => Math.round(new Date().getTime() / 1000);
 
+    // If pollingDelay === 0 then the bot is running in serverless mode and should send a `debug` level log.
+    // Else, if running in loop mode (pollingDelay != 0), then it should send a `info` level log.
+    logger[pollingDelay === 0 ? "debug" : "info"]({
+      at: "Disputer#index",
+      message: "Disputer started🔎",
+      empAddress,
+      pollingDelay,
+      errorRetries,
+      errorRetriesTimeout,
+      priceFeedConfig,
+      disputerConfig,
+      disputerOverridePrice
+    });
+
     // Load unlocked web3 accounts and get the networkId.
-    const [accounts, networkId] = await Promise.all([web3.eth.getAccounts(), web3.eth.net.getId()]);
+    const [detectedContract, accounts, networkId] = await Promise.all([
+      findContractVersion(empAddress, web3),
+      web3.eth.getAccounts(),
+      web3.eth.net.getId()
+    ]);
+
+    // Append the contract version and type to the disputerConfig, if the disputerConfig does not already contain one.
+    if (!disputerConfig) disputerConfig = {};
```

Comment:
this is the same as in the liquidator pretty much. could refactor this out at some later point but for now it'll sit fine here.

Reply:
say you want to deploy a custom bot version that the bot cant auto-detect. in this case, you would want to tell it what version to use. Also the `disputerConfig`, like the `monitorConfig` and `liquidatorConfig` are used for more things than just the contract versions. if you fed in extra settings it would come in with this object.